launchdarkly/resource_project.go

```diff
@@ -0,0 +1,115 @@
+package launchdarkly
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceProject() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceProjectCreate,
+		Read:   resourceProjectRead,
+		Update: resourceProjectUpdate,
+		Delete: resourceProjectDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"key": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceProjectCreate(d *schema.ResourceData, m interface{}) error {
+	client := m.(Client)
+
+	name := d.Get("name").(string)
+	key := d.Get("key").(string)
+
+	payload := map[string]string{
+		"name": name,
+		"key":  key,
+	}
+
+	err := client.Post("/projects", payload, map[int]bool{201: true}, nil)
+	if err != nil {
+		return err
+	}
+
+	// Default environments will be created, we want to get rid of those
```

Comment:
Why not restore the default environment instead of creating a dummy when there is no environment left. You would not need to create a dummy and delete the default here.

Reply:
I can't easily "restore" the default envs... Also they would end up acting as dummy environments anyway so I think it's simpler to use an explicit one (easier to delete it when I no longer need it afterwards).

IRL that dummy env should not exist for a long time, unless someone configures a project with this provider that has no environment.

Of course it'd be much simpler if the underlying platform accepted a project with no env, but 🤷‍♀️